### DISCUSSION TITLE
HTML Audio Element Errors in Source After Publishing

### DISCUSSION THREAD

**Discussion by NateBarham** [community/community#16659] [Projects and Issues]
So I’m trying to get the html audio element to work properly on my GitHub Pages site. In my Markdown document, I enter:

```
<audio controls>
<source src=“/assets/audio/CharacterDemo_NateBarham.mp3” type=“audio/mpeg”>
Your browser does not support the audio element.
</audio>
```

But in the source for the published page, I get:

```
<audio controls="">
&lt;source src=“/assets/audio/CharacterDemo_NateBarham.mp3” type=“audio/mpeg”&gt;
Your browser does not support the audio element.
</audio>
```

I’ve tried adding spaces between the lines, copying and pasting (and retyping) the <>’s, but every time I get the `&lt;` replacing < and `&gt;` replacing > as well as the added =“” after controls (which I’m not sure matters). The end result is a visible player with no content to load.

Any ideas? Is this a bug?

**Comment by MathiasReker** [comment]
`controls` is an empty attribute. It is not matter if it is equal to an empty string or not. The browser will understand it in the same way. :-)

**Reply by NateBarham** [reply]
Thanks! But that isn’t the problem, it’s the second line that breaks the player. I think it’s a bug in the Markdown conversion.

**Comment by manasesjesus** [comment]
it seems that the plugin doesn't like html tags in your markdown.
I took a look at your repo and the markdown plugin calls an html file that has the audio:

```md
- [Character Demo](/assets/audio/CharacterDemo_NateBarham.mp3) 
{% include playerCharacterDemo.html %}
``` 
**playerCharacterDemo.html**
```html
<p style="padding-left: 10%;">
<audio controls preload="none">
  <source src="/assets/audio/CharacterDemo_NateBarham.mp3" type="audio/mpeg">
  Your browser does not support the audio element.
</audio>
</p>
```
So, you may try adding the whole snippet, but it'd probably not work inside of a markdown list...

If you want to use it as markdown and in GitHub, you may consider using MOV or MP4 files instead. Choose a background image for your video and use your audios, like this:

https://user-images.githubusercontent.com/24204142/177369053-1e720715-61af-446e-a730-579416495922.mp4

the downside is that it renders big, so you may need to style it with the plugin.

**Comment by manasesjesus** [comment]
here's another thing you can try. Markdown plugins can be "sensitive" to characters, so these `“ ”` are not the same as these `" "`. I just noticed that in your snippet:

incorrect:
```
<source src=“/assets/audio/CharacterDemo_NateBarham.mp3” type=“audio/mpeg”>
```
correct:
```
<source src="/assets/audio/CharacterDemo_NateBarham.mp3" type="audio/mpeg">
```

**Reply by NateBarham** [reply]
I see. Hadn’t thought of this. That might be it. I’ve since figured out a workaround that I’m ok with, but I appreciate the comments here to try to get it working how I wanted originally.
